docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,432 @@
+For this tutorial, you'll define a pose for the gripper that is just above the table on one side of the tissue box, and another pose on the other side of the tissue box.
+
+You'll also want to define some waypoints.
+The Motion Service has the mathematical ability to plan complex motion from one position to another, even around an obstacle.
+However, this is more computationally intensive than planning a linear path from one point to another.
+To increase your program's efficiency, add waypoints such that the path between any two consecutive points can be linear, without intersecting the tissue box or the table:
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a start pose, where the cup starts between the gripper's jaws, on the table
+# Start pose has Z of 90mm to grab partway down the 120mm tall cup
+start_pose = Pose(x=320, y=240, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+start_pose_in_frame = PoseInFrame(reference_frame="world", pose=start_pose)
+
+# Create waypoints to increase efficiency of motion planning
+way1_pose = Pose(x=300, y=240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way1_pose_in_frame = PoseInFrame(reference_frame="world", pose=way1_pose)
+way2_pose = Pose(x=300, y=-240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way2_pose_in_frame = PoseInFrame(reference_frame="world", pose=way2_pose)
+
+# Create a pose where the cup will be set down
+end_pose = Pose(x=300, y=-250, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+end_pose_in_frame = PoseInFrame(reference_frame="world", pose=end_pose)
+```
+
+Though we are passing these waypoints in manually, the Motion Service will throw an error if we accidentally pass in a pose that would cause the arm or gripper to hit any obstacles.
+
+Note that the orientations of all the poses are the same.
+If we changed the orientation along the way, we might spill the tea!
+
+{{< alert title="Tip" color="tip" >}}
+You may be wondering how the orientations of the poses are determined.
+Our example gripper's frame is defined such that its orientation vector points from its "wrist" to the tip of its jaws.
+In the example code above, all poses have an [orientation vector](../../../internals/orientation-vector/) pointing along the positive X axis of the world frame, which is a horizontal orientation pointing "forwards" with respect to the xArm 6 base.
+When we tell the gripper to move to such a pose, its orientation vector moves to align with the orientation vector of the pose, so its jaws end up pointing along the global X axis, "forwards" from the robot base.
+This puts it in a good position for picking up and moving a cup.
+
+Additionally, for our gripper, setting `theta=0` about this particular orientation vector orients the gripper such that its jaws open and close horizontally.
+If we changed it to `theta=90` or `theta=270`, the gripper jaws would open vertically, not ideal for picking up a cup!
+{{< /alert >}}
```

Comment:
This tip is a bit lengthy, and conveys some info that I think would be helpful to all users, not just curious tip-reading ones (namely the `orientation-vector` link). What do you think about including its most important bits inline, maybe something like (replacing lines 262-274):

> Note that all poses use the same [orientation vector](../../../internals/orientation-vector/), pointing along the positive X axis of the world frame, which is a horizontal orientation pointing "forwards" with respect to the xArm 6 base.
> When we tell the gripper to move to each pose, its orientation vector moves to align with the orientation vector of the pose, so that its jaws end up pointing along the global X axis, "forwards" from the robot base.
> We also set `theta=0` for each orientation vector to orient the gripper such that its jaws open and close horizontally.
> Together, these configurations put the gripper in a good position for picking up and moving the cup. 

I removed the reference to spilling the tea because we discuss that in the next section, but I could see it retained here also if you prefer.

Reply:
(Sidebar: now that I realize "info" is a thing, if this remains as an alert I think "info" makes more sense than "tip").

I'm not entirely sure what to do here for a couple of reasons:

Some of it is specific to my particular setup. For example, theta=0 doesn't magically make the gripper orient a certain way--it totally depends on someone's frames/hardware etc.

The reason I included the other theta options is because I think it helps understand what theta does, and isn't so much about changing theta during motion and/or spilling tea as it is about understanding the concept of theta in general.

All of this information could theoretically be deduced by the reader without it being provided here, which is why I don't necessarily think it belongs in the body text. But, I added it here, because I think orientation vectors are really confusing and wanted to provide another example with explanation in case someone finds it useful.

Also, it's not necessary that all orientation vectors be the exact same, so I don't want to lead with that per se.

Reply:
Gotcha! If portions are implementation-specific, that makes a good case for being in a tip | info, agreed. I think I'd mostly just like to see the concept & link for "orientation vector" make it into the body content, if possible.

Reply:
Hmm we do talk about frames a lot in this tutorial and the previous tutorial, with links to the frame system page from both tutorials. The frame system page contains multiple mentions and links to orientation vector, so it might be okay?